resources.container.registry.private.endpoints.tf

```diff
@@ -5,35 +5,24 @@
 # Private Link for Sql - Default is "false" 
 #---------------------------------------------------------
 data "azurerm_virtual_network" "vnet" {
-  count               = var.enable_private_endpoint && var.existing_vnet_id == null ? 1 : 0
+  count               = var.enable_private_endpoint && var.virtual_network_name != null ? 1 : 0
   name                = var.virtual_network_name
   resource_group_name = local.resource_group_name
 }
 
-resource "azurerm_subnet" "snet_ep" {
-  count                                     = var.enable_private_endpoint && var.existing_subnet_id == null ? 1 : 0
-  name                                      = "snet-endpoint-${local.location}"
-  resource_group_name                       = var.existing_vnet_id == null ? data.azurerm_virtual_network.vnet.0.resource_group_name : element(split("/", var.existing_vnet_id), 4)
-  virtual_network_name                      = var.existing_vnet_id == null ? data.azurerm_virtual_network.vnet.0.name : element(split("/", var.existing_vnet_id), 8)
-  address_prefixes                          = var.private_subnet_address_prefix
-  private_endpoint_network_policies_enabled = true
-}
-
-resource "azurerm_subnet" "snet_pep" {
-  count                                     = var.create_private_endpoint_subnet && var.enable_private_endpoint ? 1 : 0
-  name                                      = "snet-private-endpoint-${var.workload_name}-${local.location}"
-  resource_group_name                       = local.resource_group_name
-  virtual_network_name                      = data.azurerm_virtual_network.vnet.0.name
-  address_prefixes                          = var.private_subnet_address_prefix
-  private_endpoint_network_policies_enabled = true
+data "azurerm_subnet" "snet" {
+  count                = var.enable_private_endpoint && var.existing_private_subnet_name != null ? 1 : 0
+  name                 = var.existing_private_subnet_name
+  virtual_network_name = azurerm_virtual_network.vnet.0.name
```

Comment:
`tf validate` is throwing an error on a clean clone.  

```
╷
│ Error: Reference to undeclared resource
│
│   on ../../resources.container.registry.private.endpoints.tf line 16, in data "azurerm_subnet" "snet":
│   16:   virtual_network_name = azurerm_virtual_network.vnet.0.name
│
│ A managed resource "azurerm_virtual_network" "vnet" has not been declared in module.acr.
│
│ Did you mean the data resource data.azurerm_virtual_network.vnet?
```

Reply:
I fixed the naming